url89cafed56a_AlexDiru_esper_compiler_final_tgz-pJ8-compiler_ProgramJ8/src/compiler/EsperParser.java

```diff
@@ -0,0 +1,1997 @@
+                case 5 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:26:13: declaration
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    pushFollow(FOLLOW_declaration_in_statement129);
+                    declaration7=declaration();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, declaration7.getTree());
+
+                    }
+                    break;
+                case 6 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:27:13: print
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    pushFollow(FOLLOW_print_in_statement144);
+                    print8=print();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, print8.getTree());
+
+                    }
+                    break;
+                case 7 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:28:13: forloop
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    pushFollow(FOLLOW_forloop_in_statement158);
+                    forloop9=forloop();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, forloop9.getTree());
+
+                    }
+                    break;
+                case 8 :
+                    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:29:13: whileloop
+                    {
+                    root_0 = (Object)adaptor.nil();
+
+
+                    pushFollow(FOLLOW_whileloop_in_statement172);
+                    whileloop10=whileloop();
+
+                    state._fsp--;
+
+                    adaptor.addChild(root_0, whileloop10.getTree());
+
+                    }
+                    break;
+
+            }
+            retval.stop = input.LT(-1);
+
+
+            retval.tree = (Object)adaptor.rulePostProcessing(root_0);
+            adaptor.setTokenBoundaries(retval.tree, retval.start, retval.stop);
+
+        }
+        catch (RecognitionException re) {
+            reportError(re);
+            recover(input,re);
+    	retval.tree = (Object)adaptor.errorNode(input, retval.start, input.LT(-1), re);
+
+        }
+
+        finally {
+        	// do for sure before leaving
+        }
+        return retval;
+    }
+    // $ANTLR end "statement"
+
+
+    public static class expr_return extends ParserRuleReturnScope {
+        Object tree;
+        public Object getTree() { return tree; }
+    };
+
+
+    // $ANTLR start "expr"
+    // C:\\Users\\Alex\\Dropbox\\EclipseWorkspace\\esper-compiler\\src\\compiler\\Esper.g:31:1: expr : ( ( PLUS | MINUS ) ^ expr expr | term | STRING );
+    public final EsperParser.expr_return expr() throws RecognitionException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 67 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.